BUILDING.md

```diff
@@ -1,152 +1,147 @@
 
 Please note that all versions after 3.12.1 will require:
 * A C++ compiler with support for C++14 or later
-
-All versions after 3.9.1 will require:
-* Qt 5.5 or later, however we advise you to use 5.7 or later
-* A C++ compiler with support for C++11 or later
+* Qt 5.15.9 later
 
 Without these or with older versions you won't be able to compile DB Browser for
-Sqlite anymore. This applies to all platforms. However, most likely you won't
+SQLite anymore.<br>This applies to all platforms. However, most likely you won't
 have to worry about these as most systems meet these requirements today.
 
-If you have any chance, please use Qt 5.7 or any later version. Even though Qt
+If you have any chance, please use Qt 5.7 or any later version.<br>Even though Qt
 5.5 and 5.6 are supported by us, there might be glitches and minor problems
 when using them.
 
+The wiki has information that is a bit more detailed or less common, but may be useful: https://github.com/sqlitebrowser/sqlitebrowser/wiki
+
 ----
-## Table of Contents
-- [Generic Linux and FreeBSD](#generic-linux-and-freebsd)
-- [Ubuntu / Debian Linux](#ubuntu--debian-linux)
-- [CentOS / Fedora Linux](#centos--fedora-linux)
-- [OpenSUSE](#opensuse)
-- [macOS](#macos)
-    - [Building a single executable binary](#building-a-single-executable-binary)
-    - [Building an .app bundle](#building-an-app-bundle)
-- [Compiling on Windows with MSVC](#compiling-on-windows-with-msvc)
+- [BUILD INSTRUCTIONS AND REQUIREMENTS](#build-instructions-and-requirements)
+  - [Linux](#linux)
+    - [Generic Linux and FreeBSD](#generic-linux-and-freebsd)
+    - [CentOS / Fedora Linux](#centos--fedora-linux)
+    - [Debian / Ubuntu Linux](#debian--ubuntu-linux)
+    - [OpenSUSE](#opensuse)
+  - [macOS](#macos)
+    - [Build an `.app` bundle](#build-an-app-bundle)
+    - [Add the extension to the app bundle (Optional)](#add-the-extension-to-the-app-bundle-optional)
+  - [Windows](#windows)
+    - [Compiling on Windows with MSVC](#compiling-on-windows-with-msvc)
     - [Cross compiling for Windows](#cross-compiling-for-windows)
 - [Build with SQLCipher support](#build-with-sqlcipher-support)
 - [Building and running the Unit Tests](#building-and-running-the-unit-tests)
-    - [Build the unit tests](#build-the-unit-tests)
-    - [Run the unit tests](#run-the-unit-tests)
+  - [Build the unit tests](#build-the-unit-tests)
+  - [Run the unit tests](#run-the-unit-tests)
 ----
 
-### Generic Linux and FreeBSD
+### Linux
+#### Generic Linux and FreeBSD
 
 The only requirements for building this code are the presence of Qt5 and
-sqlite3. Qt can be included as a static or shared library, depending on the
+SQLite 3.<br>Qt can be included as a static or shared library, depending on the
 current Qt configuration on the building machine.
 
 Provided you have Qt and cmake installed and configured, simply run:
 
-    $ cmake .
+    cmake .
```

Comment:
Hmmm, I'm conflicted here.  The `$` at the start of these is supposed to mean "compile as a non-root user".

That's less important in modern times though, where the root user is commonly used in containers and such rather than special purpose user accounts. :confused:

Reply:
The prompt symbol has been removed because it requires an unnecessary "string editing action" if someone copies and pastes this article. However, if it's important to distinguish between a normal-user and the root account, we should probably add it back. :)

Reply:
It's a tough choice, because there's not clear "correct" approach. :wink:

Looking at one of the larger sections of commands, when the `$ ` is removed it's not clear what is a command... and what is the output from the command:

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/406299/0fb26fb5-a0dd-4542-9b32-3d5abeef1974)

With `$ ` added back to those, it would be clear they are all commands.

But on the other hand, it's easier to copy-n-paste when the `$ ` aren't there. :wink:

Meh, I'll leave it up to you to choose.  I'm ok with either approach. :stuck_out_tongue_winking_eye:

Reply:
Hmm, not an easy choice. In fact, there are places where I've kept the prompt symbols for the same reason.

The other commands are short, so I don't think it's hard to tell if they're commands or output.
But the code you attached, yes, it's confusing even to me. On the other hand, the code you attached is long and complicated, so it's better to copy and paste. Why don't we just add a note for that part?

> **Note**: The following series of statements that look a bit complicated are all commands.

Like this?

Reply:
> Like this?

Sounds reasonable. :smile:

Just had an alternative idea for this particular block of code, that might also be safer as well.

We could do something like this for this section:

```
/opt/homebrew/opt/db4sqtub@5/bin/macdeployqt DB\ Browser\ for\ SQLite.app && \
mkdir build/DB\ Browser\ for\ SQLite.app/Contents/Extensions && \
clang -I /opt/homebrew/opt/db4ssqliteftsub@5/include -L /opt/homebrew/opt/db4ssqliteftsub@5/lib \
  -fno-common -dynamiclib src/extensions/extension-formats.c -o formats-arm64.dylib && \
arch -x86_64 clang -I /opt/homebrew/opt/db4ssqliteftsub@5/include \
  -L /opt/homebrew/opt/db4ssqliteftsub@5/lib -fno-common -dynamiclib \
  src/extensions/extension-formats.c -o formats-x86_64.dylib && \
lipo -create -output build/DB\ Browser\ for\ SQLite.app/Contents/Extensions/formats.dylib \
  formats-arm64.dylib formats-x86_64.dylib && \
install_name_tool -id "@executable_path/../Extensions/formats.dylib" \
  build/DB\ Browser\ for\ SQLite.app/Contents/Extensions/formats.dylib && \
ln -s formats.dylib build/DB\ Browser\ for\ SQLite.app/Contents/Extensions/formats.dylib.dylib
```

By that, I'm meaning:

* We break apart the command lines using `\`, so they fit into the GitHub code display without people needing to scroll
  * That also makes it a bit easier to tell these are commands
* We use `&&` between the commands, so the subsequent commands only run if the earlier ones are successful
  * I think this would be "safer", in that if an error happens then the following commands won't try running anyway

If that's too complicated though, lets just go with your suggestion. :smile:

Reply:
That's a good idea. but it seems a bit complicated to me.
Why not keep it for now and provide an option in CMake in the future (like `make extension`)
or perhaps a separate script?

Reply:
Ahhh.  Yeah, we could do that too. :smile:

Reply:
k, lets go with your idea, with a slight adjustment for extra clarity:

> Note: The following lines look a bit complicated, but are all commands that you need to run.

Reply:
Thank you again for your kindness and valuable review. We are now ready to merge it.

Reply:
> k, lets go with your idea, with a slight adjustment for extra clarity:

Oh wait, this comment has now appeared. Please be patient :)

Reply:
Resolve via [4a06ba0](https://github.com/sqlitebrowser/sqlitebrowser/pull/3485/commits/4a06ba003e10fd4ba5e8f1b37b6792a47d61f7c9)